contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/hbm/BaseReviewServiceImpl.java

```diff
@@ -222,6 +222,24 @@ public ContentReviewItem getFirstItemByContentId(String contentId) {
 		return existingItems.get(0);
 	}
 	
+	public ContentReviewItem getFirstItemByExternalId(String externalId) {
+		//due to the impossibility to get the right paper id from the turnitin callback
+		//we need to get the paper id associated to the original submission
+		Search search = new Search();
+		search.addRestriction(new Restriction("externalId", externalId));
```

Comment:
If you are coping with the case of there being multiple items with the same externalId then you should add an order by clause so that the results are repeatable rather than hoping the DB always returns them in the same order.